File: script-stubs/helpers/helperFunctions.py
import json
import sys

# Read inputs, calling script shoud assign the return value to a variable
def biab_inputs():
	inputFile = open(sys.argv[1] + '/input.json')
	return json.load(inputFile)

# Add outputs throughout the script
def biab_output(key, value):
	biab_output_list[ key ] = value
	print("Output added for \"", key, "\"\n")

# Non-breaking messages
def biab_info(message):
	biab_output("info", message)

def biab_warning(message):
	biab_output("warning", message)

# Output error message and stop the pipeline
def biab_error_stop(errorMessage):
	biab_output_list[ "error" ] = errorMessage
	sys.exit(errorMessage)

File: script-stubs/helpers/helperFunctions.R
options(error=traceback, keep.source=TRUE, show.error.locations=TRUE)

# Define repo for install.packages
repositories = getOption("repos")
repositories["CRAN"] = "https://cloud.r-project.org/"
options(repos = repositories)

library(rjson)

# Read inputs, calling script should assign the return value to a variable
biab_inputs <- function(){
    fromJSON(file=file.path(outputFolder, "input.json"))
}

# Add outputs throughout the script
biab_output <- function(key, value){
    biab_output_list[[ key ]] <<- value
    cat("Output added for \"", key, "\"\n")
}

# Non-breaking messages
biab_info <- function(message) biab_output("info", message)
biab_warning <- function(message) biab_output("warning", message)

# Output error message and stop the pipeline
biab_error_stop <- function(errorMessage){
    biab_output_list[[ "error" ]] <<- errorMessage
    stop(errorMessage)
}